### PR TITLE
Implement indent guides for the current set of nodes we have outlining for in C# and VB.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide @Pilchie @natidea 

This is to have Indent Guides on for preview5 for C# and VB.

**Comment by CyrusNajmabadi** [comment]
Have tested out VB and C#.  Both look as we expect for the current state of indent guides.

File: src/EditorFeatures/Next/Structure/VisualStudio15StructureTaggerProvider.cs [hunk 1]
```diff
@@ -49,31 +49,90 @@ public override int GetHashCode(IBlockTag obj)
...
+                    case BlockTypes.Loop: return PredefinedStructureTypes.Loop;
+                    case BlockTypes.TryCatchFinally: return PredefinedStructureTypes.TryCatchFinally;
+                    case BlockTypes.Standalone: return PredefinedStructureTypes.Standalone;
+
+                    // Expressions
+                    case BlockTypes.AnonymousMethod: return PredefinedStructureTypes.AnonymousMethodBlock;
+
+                    // These types don't currently map to any editor types.  Just make them
+                    // the 'Unknown' type for now.
+                    case BlockTypes.Enum:
```
Comment (Pilchie) [inline src/EditorFeatures/Next/Structure/VisualStudio15StructureTaggerProvider.cs]: Better to map this to class or struct instead?

**Comment by Pilchie** [comment]
I withdraw that - this only touches editorfeatures.next, which is already loaded for completion...

**Comment by Pilchie** [comment]
@dotnet-bot test vsi please

**Comment by Pilchie** [comment]
Does this cause any breaking changes for TS/JS?

**Comment by CyrusNajmabadi** [comment]
retest windows_vsi_p3_open_prtest please

**Comment by CyrusNajmabadi** [comment]
No, this should cause no issues for TS/JS.  We're just adding the 'type' information into the BlockSpan so that we get Indent-Guides.

Adding @paulvanbrenk and @vladima for visibility though.

**Comment by CyrusNajmabadi** [comment]
Tagging @natidea since this is for Preview5.

**Comment by CyrusNajmabadi** [comment]
Do we need to tag Matt?

**Comment by natidea** [comment]
Tagging @MattGertz 
dev15-preview-5 is effectively in escrow (assuming we RI today) so this change will need to be approved by shiproom before it can be merged.

**Comment by MattGertz** [comment]
@CyrusNajmabadi Risk assessment?  I would definitely like to get this in for Escrow.

**Comment by CyrusNajmabadi** [comment]
> Risk assessment

Low.

**Comment by MattGertz** [comment]
Approved to take to Escrow pending test completion.

File: src/EditorFeatures/Core/Implementation/Structure/RoslynOutliningRegionTag.cs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            _projectionBufferFactoryService = projectionBufferFactoryService;
+            _editorOptionsFactoryService = editorOptionsFactoryService;
+            _subjectBuffer = snapshot.TextBuffer;
+            BlockSpan = outliningSpan;
+
+            _hintSpan = snapshot.CreateTrackingSpan(BlockSpan.HintSpan.ToSpan(), SpanTrackingMode.EdgeExclusive);
+        }
+
+        public object CollapsedHintForm =>
+            new ViewHostingControl(CreateElisionBufferView, CreateElisionBuffer);
```
Comment (DustinCampbell) [inline src/EditorFeatures/Core/Implementation/Structure/RoslynOutliningRegionTag.cs]: Kinda hate that this expensive allocation is a property getter. I suppose its defined that way on IOutliningRegionTag?

File: src/Features/Core/Portable/Structure/BlockTypes.cs [hunk 3]
```diff
@@ -4,25 +4,40 @@ namespace Microsoft.CodeAnalysis.Structure
...
-        public const string Operator = "Operator";
-        public const string PropertyBlock = "PropertyBlock";
-        public const string Standalone = "Standalone";
-        public const string Struct = "Struct";
-        public const string Structural = "Structural";
-        public const string TryCatchFinally = "TryCatchFinally";
-        public const string Unknown = "Unknown";
+        // Basic types.
+        public const string Structural = nameof(Structural);
+        public const string Nonstructural = nameof(Nonstructural);
```
Comment (DustinCampbell) [inline src/Features/Core/Portable/Structure/BlockTypes.cs]: Casing seems weird to me. Should this actually be `NonStructural`?

File: src/EditorFeatures/Core/Implementation/Structure/RoslynOutliningRegionTag.cs (hunk 2 above)
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/Structure/RoslynOutliningRegionTag.cs]: Yup.  It's an interface method.  And the editor only calls it when neeed.  We def do not want to eagerly create it.

File: src/Features/Core/Portable/Structure/BlockTypes.cs (hunk 3 above)
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Structure/BlockTypes.cs]: I made it this way to match the editor side.  They actually started as NonStructural but moved ot Nonstructural.

**Comment by CyrusNajmabadi** [comment]
retest windows_vsi_p1_open_prtest  please

**Comment by CyrusNajmabadi** [comment]
retest windows_vsi_p3_open_prtest please

**Comment by CyrusNajmabadi** [comment]
retest windows_vsi_p1_open_prtest  please

**Comment by CyrusNajmabadi** [comment]
retest windows_vsi_p1_open_prtest please

**Comment by CyrusNajmabadi** [comment]
retest windows_vsi_p3_open_prtest  please

**Comment by CyrusNajmabadi** [comment]
retest windows_vsi_p3_open_prtest please

**Comment by CyrusNajmabadi** [comment]
retest windows_vsi_p1_open_prtest please

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit64_prtest please

**Pull request merged**